pkg/scanners/clairify/clairify.go

```diff
@@ -411,12 +421,38 @@ func (c *clairify) GetVulnerabilities(image *storage.Image, components *clairGRP
 	return convertImageToImageScan(image.GetMetadata(), resp.GetImage()), nil
 }
 
+func callGRPCWithRetryOnError(ctx context.Context, operation string, f func() error) error {
+	err := wait.ExponentialBackoffWithContext(ctx, scannerDefaultBackoff, func() (bool, error) {
+		err := f()
+		if err != nil {
+			e, _ := status.FromError(err)
+			switch e.Code() {
+			case codes.Internal, codes.Aborted, codes.Unavailable, codes.Unknown:
```

Comment:
Hmmm do you know when we get internal or unknown? Is this common, do we do proper error handling from scanner side?

Reply:
I picking up these codes based on [their description](https://pkg.go.dev/google.golang.org/grpc/codes). 

`codes.Unavailable` and `codes.Aborted` are recommended for retries, see `FailedPrecondition` comment on the litmus test. I am in favor of keeping the first two.

`codes.Unknown` happens when the reply cannot be parsed properly, it's a catch all scenario and I agree we might not need to retry on "unknown".

Finally `codes.Internal` is what [scanner actually returns](https://github.com/stackrox/scanner/blob/10e708a2ca778683cee3ede3926cad3581a2ae53/api/v1/nodescan/service.go#L320). But it seems Scanner simply wrap all errors, so invalid input and things that we should not retry will be returned as `codes.Internal`. I vote we should remove it.